.github/workflows/ci.yml

```diff
@@ -0,0 +1,81 @@
+# Continuous Integration Workflow to check pushes to master and pull requests:
+# - code style guidelines are followed (make check-style)
+# - tests pass (make test)
+# - code builds (make dist)
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  ci:
+    name: Check style; build; test
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Go v1.18.x
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.18.x
+
+      - name: Set up Node.js v14.x
+        uses: actions/setup-node@v1
+        with:
+          node-version: 14.x
+
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Install Yarn
+        run: npm install -g yarn
+
+      - name: Install golangci
+        run: curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.52.2
+
+      - name: Cache go modules
+        uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Cache node modules
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
```

Comment:
is this correct? because we are using `yarn`

Reply:
updating the job with yarn.lock